### PR TITLE
Team edit and more

### PR DESCRIPTION
**Pull request by abp6318**
- [x] Fix Team edit, form looks okay on frontend, but on submission, backend crashes -- reason unknown -- might need to unfold `res` bits
- [x] Deleting a team deletes all teamIDs from users
- [x] Deleting a user deletes them from the teams
- [x] Adding a user to a team updates the team array (reverse of edit team) --> Vicky and I tried this and IT SUCKS SO HARD OH GOD
- [x] Deleting a university needs to cascade delete teams, players, and games
- [x] Remove ability to update university's ID
- [x] Set user's teamID to null if they are removed from a team that is being edited

To test:
- [x] Team create, edit, and delete (check cascades)
- [x] University create, edit, and delete (check cascades & can't edit universityID)

### CODE REVIEW

File: routes/user/userSecure.js [hunk 1]
```diff
@@ -196,13 +197,19 @@ router.put('/', async (req, res) => {
                 if (updUser) {
                     await UserInfo.updateOne({_id: updUser._id}, req.body)
                     .then(async function (data, err){
+                        const teamToAdd = async () => {
+                            await TeamInfo.updateOne({_id: req.body.teamID}, {$push: {players: updUser._id}});
+                        }
+                        teamToAdd();
```
Comment (abp6318) [inline routes/user/userSecure.js:203]: does this need await?

File: routes/user/userSecure.js [hunk 2]
```diff
@@ -196,13 +197,19 @@ router.put('/', async (req, res) => {
...
                         if (err) {
-                            res.status(500).json(err);
+                            res.status(500).json(err+"");
                         }
                         else {
                             const updated = await UserInfo.findOne({_id: ObjectId(req.body.id)});
                             res.status(200).json({updated});
                         }
+                        //await addUserIdToTeam();
```
Comment (abp6318) [inline routes/user/userSecure.js]: was this leftover from something else?

File: routes/user/userSecure.js (hunk 1 above)
Comment (vsoler27) [reply routes/user/userSecure.js:203]: literally no clue but it worked? I thought it did but maybe not?

Comment (vsoler27) [reply routes/user/userSecure.js:203]: "no it does" - aidan

**Review by vsoler27 — APPROVED** [review]
yes maam

**Pull request merged**